develop/api_guc.rst

```diff
@@ -457,6 +457,31 @@ time.
 For long queries (those taking >500ms), slow start might add latency, but for
 short queries it's faster. The default value is 10ms.
 
+citus.max_shared_pool_size (integer)
+************************************
+
+Specifies the maximum number of connections that the coordinator node, across
+all simultaneous sessions, is allowed to make per worker node. PostgreSQL must
+allocate fixed resources for every connection and this GUC helps ease
+connection pressure on workers.
+
+Without connection throttling, every multi-shard query creates connections on
+each worker proportional to the number of shards it accesses (in particular,
+#shards/#workers). Running dozens of multi-shard queries at once can easily hit
+worker nodes' ``max_connections`` limit, causing queries to fail.
+
+The default value, 0, caps the connections to the coordinator's own
```

Comment:
Minor note:  I think we don't need to explicitly mention about the default value `0`, as that is  an internal value, the users don't  see it  at all. Because, we override the value when showing  to the users: https://github.com/citusdata/citus/blob/c6f5d5fe885b35d6ac7fb865db30222c945b3d49/src/backend/distributed/shared_library_init.c#L1595-L1612

```SQL
postgres=# show max_connections ;
┌─────────────────┐
│ max_connections │
├─────────────────┤
│ 100             │
└─────────────────┘
(1 row)

Time: 0.419 ms
postgres=# show citus.max_shared_pool_size ;
┌────────────────────────────┐
│ citus.max_shared_pool_size │
├────────────────────────────┤
│ 100                        │
└────────────────────────────┘
(1 row)

Time: 0.549 ms
postgres=# 

``` 
So, maybe  `The default value, 0, ` -> `By default, the value is automatically set equal to the coordinator's own ...`